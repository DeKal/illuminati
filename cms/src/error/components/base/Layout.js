import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import Header from 'error/components/base/Header'
import Title from 'error/components/base/Title'
import Description from 'error/components/base/Description'
import Button from 'error/components/base/Button'

const Layout = ({ title, description, detail }) => {
  return (
    <Fragment>
      <Header>Oops!</Header>
      <Title>{title}</Title>
      <Description>{description}</Description>
      {detail && (
        <Description style={{ color: 'red' }}>Error: {detail}</Description>
      )}
      <a href="/cms">
        <Button> Go To Homepage </Button>
      </a>
    </Fragment>
  )
}

Layout.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  detail: PropTypes.string
}

export default Layout
