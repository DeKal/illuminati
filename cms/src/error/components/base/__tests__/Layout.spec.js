import * as React from 'react'
import { shallow } from 'enzyme'
import Layout from '../Layout'

describe('Layout', () => {
  it('should render correctly', function() {
    const wrap = shallow(
      <Layout
        title="Error"
        description="Internal server error"
        detail="Error at line xxx"
      />
    )
    expect(wrap).toMatchSnapshot()
  })
})
