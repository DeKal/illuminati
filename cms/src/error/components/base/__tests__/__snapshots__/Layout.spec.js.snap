// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<Fragment>
  <Header>
    Oops!
  </Header>
  <Styled(h2)>
    Error
  </Styled(h2)>
  <Styled(p)>
    Internal server error
  </Styled(p)>
  <Styled(p)
    style={
      Object {
        "color": "red",
      }
    }
  >
    Error: 
    Error at line xxx
  </Styled(p)>
  <a
    href="/cms"
  >
    <Styled(a)>
       Go To Homepage 
    </Styled(a)>
  </a>
</Fragment>
`;
